src/_pytest/runner.py

```diff
@@ -85,19 +85,17 @@ def runtestprotocol(item, log=True, nextitem=None):
     if hasrequest and not item._request:
         item._initrequest()
     rep = call_and_report(item, "setup", log)
-    reports = [rep]
     if rep.passed:
         if item.config.getoption("setupshow", False):
             show_test_item(item)
         if not item.config.getoption("setuponly", False):
-            reports.append(call_and_report(item, "call", log))
-    reports.append(call_and_report(item, "teardown", log, nextitem=nextitem))
+            call_and_report(item, "call", log)
+    call_and_report(item, "teardown", log, nextitem=nextitem)
```

Comment:
Probably there are plugins which might be using this. Although not public API, we shouldn't remove this without a good reason I think.

Reply:
Indeed - https://github.com/pytest-dev/pytest-forked/blob/42d7146d65749b54571480db51a8461aa509b3ac/src/pytest_forked/__init__.py#L50.